x-pack/plugins/enterprise_search/public/applications/shared/enterprise_search_url/external_url.ts

```diff
@@ -27,12 +27,16 @@ export const externalUrl = {
   },
 };
 
-export const getEnterpriseSearchUrl = (path: string = ''): string => {
-  return externalUrl.enterpriseSearchUrl + path;
+export const getEnterpriseSearchUrl = (
+  path: string = '',
+  namespace: string = '',
+  useAnchor: boolean = false
+): string => {
+  return `${externalUrl.enterpriseSearchUrl}${namespace}${(useAnchor && '#') || ''}${path}`;
 };
-export const getAppSearchUrl = (path: string = ''): string => {
-  return getEnterpriseSearchUrl('/as' + path);
+export const getAppSearchUrl = (path: string = '', useAnchor: boolean = false): string => {
+  return getEnterpriseSearchUrl(path, '/as', useAnchor);
```

Comment:
```suggestion
export const getAppSearchUrl = (path: string = ''): string => {
  return getEnterpriseSearchUrl(path, '/as#');
```

TBH, I'm not a huge fan of how our App Search links work and how some of them require `#` anchors and some don't (because they correctly redirect).

So if I had a preferred option between the two, I'd probably choose this one for simplicity, since EntSearch fixed login redirects not preserving hashes a while back.